CONTRIBUTING.md

```diff
@@ -0,0 +1,45 @@
+# Contributing to leaderboard-backend
+We appreciate your help!
+
+## Before filing an issue
+If you are unsure whether you have found a bug, please consider asking in [our discord](https://discord.gg/TZvfau25Vb) first.
+
+Similarly, if you have a question about a potential feature, [the discord](https://discord.gg/TZvfau25Vb) can be a fantastic resource for first comments.
+
+## Filing issues
+Filing issues is as simple as going to [the issue tracker](https://github.com/speedrun-website/leaderboard-backend/issues), and adding an issue!
+
+## Contributing code
+### Example code contribution flow
+1. Make a fork of this repo.
+1. Name a branch on your fork something descriptive for this change (eg. `UpdateMakefile`).
+1. Commit your changes (Tip! Please read our [Style guide](#style-guide) to help the pull request process go smoothly).
+1. Verify your changes work with `make test`.
+1. Push your branch.
+1. Open a pull request to `speedrun-website/leaderboard-backend`.
+1. Get your pull request approved.
+1. Get someone to click `Rebase and merge`.
+1. Celebrate your amazing changes! 🎉
+
+## Style guide
+### General
+- Be inclusive, this is a project for everyone.
+- Be descriptive, it can be hard to understand abbreviations or short-hand.
+
+### GoLang
+- Comments should be full sentences, starting with a capital letter and ending with punctuation.
+- Comments above a func or struct should start with the name of the thing being described.
+- Wrap errors before returning with `oops.Wrapf(` to help build useful stacks for debugging.
+- Early returns are great, they help reduce nesting!
+- Avoid `interface{}` where possible, if we need such a generic please add a comment explaining what the real type is.
+
+### Git
+- Prefer branch names without slashes.
+  - Nested paths can act strange when other people start looking at your branch.
```

Comment:
I'm probably missing something but how much does this even matter? Wouldn't the branch name not matter once it's merged anyway?

Reply:
I was actually including this from https://github.com/speedrun-website/leaderboard-backend/issues/32#issuecomment-878578260.

Reply:
The explicit name standard does side-step this issue, which is nice. I am not sure we can really need to enforce this super hard anyways, since the branch names basically disappear.

Reply:
Having a naming standard is just very nice for when the PR comes in - very easy to tell what something relates to outside of the context of the actual Issue page. Will help readability down the line when we don't just have a single branch to merge into for potential beta features down the line that shouldn't necessarily go right into `main`.

Reply:
I can update this to suggest a format and to list two things that are not allowed (main, /s). Does that sound good?

Reply:
Sounds good!

Reply:
Updated!